src/lib/onlyText.ts

```diff
@@ -22,9 +21,14 @@ const onlyText = (children: ReactNode | ReactNode[]): string => {
   return Children.toArray(children).reduce((text: string, child: ReactNode): string => {
     let newText = '';
 
-    if (isValidElement(child) && hasChildren(child)) {
+    const isValidElementResult = isValidElement<{ children?: React.ReactNode | React.ReactNode[] }>(
+      child,
+    );
+    const hasChildren = isValidElementResult && 'children' in child.props;
+
+    if (isValidElementResult && hasChildren) {
       newText = onlyText(child.props.children);
-    } else if (isValidElement(child) && !hasChildren(child)) {
```

Comment:
Since the `else if` already implies `!hasChildren(child)` this should only be now `isValidElement(child)`

Reply:
plz, see my previous comment